File: 1502.判断能否形成等差数列.js
/*
 * @lc app=leetcode.cn id=1502 lang=javascript
 *
 * [1502] 判断能否形成等差数列
 */

// @lc code=start
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var canMakeArithmeticProgression = function(arr) {
    arr.sort((a, b) => a - b);
    let dif = arr[1] - arr[0];
    let n = arr.length;
    for (let i = 1; i < n - 1; i++) {
        if (arr[i + 1] - arr[i] !== dif) {
            return false;
        }
    }
    return true;
};
// @lc code=end

File: 1502.判断能否形成等差数列.java
/*
 * @lc app=leetcode.cn id=1502 lang=java
 *
 * [1502] 判断能否形成等差数列
 */

// @lc code=start
class Solution {
    public boolean canMakeArithmeticProgression(int[] arr) {
        Arrays.sort(arr);
        int diff = arr[1] - arr[0];
        int n = arr.length;
        for (int i = 1; i < n - 1; i++) {
            if (arr[i + 1] - arr[i] != diff) {
                return false;
            }
        }
        return true;
    }
}
// @lc code=end

